gss-cb-signalling.xml

```diff
@@ -186,21 +186,19 @@ Whenever both the initiator and the acceptor provide matching channel bindings t
           </t>
         </section>
       </section>
-      <section title="New Mechanism Attributes" anchor="sub_New_Mechanism_Attributes">
+      <section title="New Mechanism Attribute" anchor="sub_New_Mechanism_Attribute">
         <t>
           <list style="symbols">
             <t>
 We add a new mechanism attribute, GSS_C_MA_CBINDING_CONFIRM, to indicate that the initiator can and always does learn whether the acceptor application supplied channel bindings (assuming mutual auth has been requested). Mechanisms advertising this attribute MUST always indicate acceptor channel bound state to the initiator.</t>
-            <t>
-We add a new mechanism attribute, GSS_C_MA_CBINDING_MAY_CONFIRM, to indicate that the initiator may learn whether the acceptor application supplied channel bindings (assuming mutual auth has been requested), but only when the acceptor implementation of the mechanism has been suitably updated. Mechanisms MUST advertise this attribute when the local initiator functionality for acceptor channel bound state indication exists but the acceptor may lack the same functionality (because, for example, the mechanism predates this specification).</t>
           </list>
         </t>
         <t>
 OID assignments TBD.</t>
       </section>
       <section title="Request Flag for Acceptor Confirmation of Channel Binding" anchor="d1e605">
         <t>
-We add a new request flag for GSS_Init_sec_context(), req_cb_confirmation_flag, to be used by initiators that insist on acceptors providing channel bindings. This flag is only of use to mechanism-negotiation pseudo-mechanisms (e.g., SPNEGO <xref target="RFC4178"/>): if set, the pseudo-mechanism MUST NOT negotiate any mechanisms that lack both the GSS_C_MA_CBINDING_CONFIRM and GSS_C_MA_CBINDING_MAY_CONFIRM mechanism attributes, and SHOULD NOT negotiate mechanisms that lack the GSS_C_MA_CBINDING_CONFIRM mechanism attribute (except if allowed by local configuration).</t>
+We add a new request flag for GSS_Init_sec_context(), req_cb_confirmation_flag, to be used by initiators that insist on acceptors providing channel bindings. This flag is only of use to mechanism-negotiation pseudo-mechanisms (e.g., SPNEGO <xref target="RFC4178"/>): if set, the pseudo-mechanism MUST NOT negotiate any mechanism which lacks the GSS_C_MA_CBINDINGS attribute and SHOULD prefer to negotiate mechanisms which have the GSS_C_MA_CBINDING_CONFIRM attribute.</t>
```

Comment:
Previously this flag would exclude any mechanism which never confirms channel bindings to the initiator; now it excludes any mechanism that doesn't support channel bindings at all.  I don't recall discussion leading to that semantics.

At this moment I'm in favor of removing the new intiiator flag altogether, absent a clear understanding of what would lead an initiating application to use the flag and how the semantics support that use case.

Reply:
I mean, it seemed like the logical behavior to me as an adaptation of the recommendation that mechs lacking CBINDING_CONFIRM shouldn't be negotiated.  I guess we could drop that part and have it use the order: CBINDING_CONFIRM, CBINDINGS, all others?

@nicowilliams, since you want this flag, what do you think about all this?